### PR TITLE
Relationships

### PR DESCRIPTION
**Pull request by italolelis**
- Adds relationship support
- Added new pkg for query builder
- Removed database.Cell which is not needed since we only need a `map[string]interface{}`

Tests were made with this configuration:

```toml
[[Tables]]
  Name = "users"
  [Tables.Filter]
    Limit = 100
    [Tables.Filter.Sorts]
      orderNr = "asc"
  [Tables.Anonymise]
    email = "EmailAddress"
    username = "UserName"

[[Tables]]
  Name = "orders"

  [Tables.Filter]
    Limit = 100

  [[Tables.Relationships]]
    ReferencedTable = "users"
    ReferencedKey = "users.id"
    ForeignKey = "orders.user_id"
```

We have to execute this command:

```sh
go run main.go steal --from=postgres://hello:fresh@localhost:8050/klepto?sslmode=disable --to=os://stdout
```

And the output to `stdout` was

```sql
INSERT INTO users (gender,id,username,email,active) VALUES ('m','0d60a85e-0b90-4482-a14c-108aea2557aa','NicoleFlores','fRice@Muxo.gov','true')
INSERT INTO users (active,gender,id,username,email) VALUES ('true','NULL','39240e9f-ae09-4e95-9fd0-a712035c8ad7','MarthaPayne','BobbySchmidt@Zoomlounge.name')
INSERT INTO users (gender,id,username,email,active) VALUES ('f','9e4de779-d6a0-44bc-a531-20cdb97178d2','BillyGarrett','BettyCarroll@Centizu.org','false')
INSERT INTO users (gender,id,username,email,active) VALUES ('m','66a45c1b-19af-4ab5-8747-1b0e2d79339d','BillyBoyd','WilliamWashington@Skibox.mil','true')
INSERT INTO orders (user_id,id) VALUES ('66a45c1b-19af-4ab5-8747-1b0e2d79339d','b9bcd5e1-75e6-412d-be87-278003519717')
INSERT INTO orders (id,user_id) VALUES ('7ee31a7f-5140-483b-8ba1-fa8f116219c0','66a45c1b-19af-4ab5-8747-1b0e2d79339d')
INSERT INTO orders (user_id,id) VALUES ('66a45c1b-19af-4ab5-8747-1b0e2d79339d','dda290ff-6243-46d9-83cb-acbad41e936e')
INSERT INTO orders (id,user_id) VALUES ('453f4498-b4e0-485f-94fa-72f233bb7958','9e4de779-d6a0-44bc-a531-20cdb97178d2')
INSERT INTO orders (id,user_id) VALUES ('8bdf39d8-616c-45d4-826f-bad30cb4e1a3','9e4de779-d6a0-44bc-a531-20cdb97178d2')
INSERT INTO orders (id,user_id) VALUES ('f1f7c9c7-bdb7-4626-a5c9-44d8942e52dd','39240e9f-ae09-4e95-9fd0-a712035c8ad7')
INSERT INTO orders (id,user_id) VALUES ('e650ad64-f1e4-4f91-abea-ec1a70992926','39240e9f-ae09-4e95-9fd0-a712035c8ad7')
```

### CODE REVIEW

File: pkg/anonymiser/anonymiser_test.go [hunk 1]
```diff
@@ -1,9 +1,10 @@
 package anonymiser
 
 import (
-	"reflect"
 	"testing"
 
+	"github.com/hellofresh/klepto/pkg/reader"
+
 	"github.com/stretchr/testify/assert"
```
Comment (rafaeljesus) [inline pkg/anonymiser/anonymiser_test.go]: Extra spaces here

File: pkg/anonymiser/anonymiser_test.go [hunk 2]
```diff
@@ -1,9 +1,10 @@
 package anonymiser
 
 import (
-	"reflect"
 	"testing"
 
+	"github.com/hellofresh/klepto/pkg/reader"
+
```
Comment (rafaeljesus) [inline pkg/anonymiser/anonymiser_test.go]: extra spaces here too

File: pkg/dumper/query/dumper.go [hunk 3]
```diff
@@ -27,96 +29,123 @@ func NewDumper(output io.Writer, rdr reader.Reader) dumper.Dumper {
...
 	return nil
 }
 
+func (d *textDumper) toSQLColumnMap(row database.Row) map[string]interface{} {
+	sqlColumnMap := make(map[string]interface{})
+
+	for column, value := range row {
+		strValue, err := d.toSQLStringValue(value)
+		if err != nil {
+			log.WithError(err).Debug("could not convert value to string")
```
Comment (rafaeljesus) [inline pkg/dumper/query/dumper.go]: Shouldn't we return if there's an error?

**Comment by rafaeljesus** [comment]
👍

**Comment by rafaeljesus** [comment]
:shipit:

File: pkg/reader/generic/sql.go [hunk 4]
```diff
@@ -17,7 +19,7 @@ type SqlReader struct {
 func (s *SqlReader) GetColumns(table string) (columns []string, err error) {
 	// TODO fix since it fails for empty tables
 	var rows *sql.Rows
-	if rows, err = s.Connection.Query(fmt.Sprintf("SELECT * FROM %s LIMIT 1", table)); err != nil {
+	if rows, err = sq.Select("*").From(table).Limit(1).RunWith(s.Connection).Query(); err != nil {
```
Comment (kieranajp) [inline pkg/reader/generic/sql.go]: Yeah this is much better

**Pull request merged**